src/main/java/de/qabel/desktop/ui/actionlog/ActionlogController.java

```diff
@@ -21,75 +34,175 @@
-		injectionContext.put("text", text);
-		MyActionlogItemView myItemView = new MyActionlogItemView(injectionContext::get);
-		messages.getChildren().add(myItemView.getView());
-		messageView.add(myItemView);
+	void receiveDropMessages(Date siceDate) throws QblDropInvalidMessageSizeException, QblVersionMismatchException, QblSpoofedSenderException {
+		List<DropMessage> dropMessages = getDropMassages(siceDate);
 
-		injectionContext = new HashMap<>();
+		if (dropMessages == null) {
+			return;
+		}
 
-		OtherTextWrapper wrapper = new OtherTextWrapper();
-		wrapper.setText(text);
-		Identity i = clientConfiguration.getSelectedIdentity();
-		Contact contact = new Contact(i, i.getAlias(), i.getDropUrls(), i.getEcPublicKey());
-		wrapper.setContact(contact);
-		injectionContext.put("wrapper", wrapper);
+		for (DropMessage d : dropMessages) {
+			addMessageToActionlog(d);
+		}
+	}
+
+	void addMessageToActionlog(DropMessage dropMessage) {
+		lastDate = dropMessage.getCreationDate();
+		Map<String, Object> injectionContext = new HashMap<>();
+		Contact sender = findSender(dropMessage);
+		injectionContext.put("dropMessage", dropMessage);
+		injectionContext.put("contact", sender);
 		OtherActionlogItemView otherItemView = new OtherActionlogItemView(injectionContext::get);
 		messages.getChildren().add(otherItemView.getView());
 		messageView.add(otherItemView);
-		text = "short Text";
-		injectionContext = new HashMap<>();
-		injectionContext.put("text", text);
-		myItemView = new MyActionlogItemView(injectionContext::get);
+	}
+
+	void addOwnMessageToActionlog(DropMessage dropMessage) {
+		Map<String, Object> injectionContext = new HashMap<>();
+		injectionContext.put("dropMessage", dropMessage);
+		MyActionlogItemView myItemView = new MyActionlogItemView(injectionContext::get);
 		messages.getChildren().add(myItemView.getView());
 		messageView.add(myItemView);
+		textarea.setText("");
+	}
 
-		wrapper = new OtherTextWrapper();
-		wrapper.setText(text);
-		i = clientConfiguration.getSelectedIdentity();
-		contact = new Contact(i, i.getAlias(), i.getDropUrls(), i.getEcPublicKey());
-		wrapper.setContact(contact);
-		injectionContext.put("wrapper", wrapper);
-		otherItemView = new OtherActionlogItemView(injectionContext::get);
-		messages.getChildren().add(otherItemView.getView());
-		messageView.add(otherItemView);
+	private Contact findSender(DropMessage dropMessage) {
 
+		List<Contact> contactList = null;
+
+		try {
+			contactList = contactRepository.findAllContactFromOneIdentity(identity);
+		} catch (EntityNotFoundExcepion entityNotFoundExcepion) {
+			entityNotFoundExcepion.printStackTrace();
```

Comment:
This code has no value. If the user expects to find something that doesn't exist, rethrowing the exception is fine. Anyways, "findAll*" doesn't have to throw an EntityNotFound exception. An Empty list would fit semantically if the identity has no contacts.